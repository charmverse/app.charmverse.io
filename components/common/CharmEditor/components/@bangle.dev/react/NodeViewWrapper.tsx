import type { NodeView, NodeViewProps } from '@bangle.dev/core';
import React from 'react';

export type RenderNodeViewsFunction = (
  props: NodeViewProps & { children: React.ReactNode },
) => React.ReactNode;

interface PropsType {
  nodeView: NodeView;
  renderNodeViews: RenderNodeViewsFunction;
  nodeViewUpdateStore: WeakMap<NodeView, () => void>;
}

interface StateType {
  nodeViewProps: NodeViewProps;
}

export class NodeViewWrapper extends React.Component<PropsType, StateType> {
  update: () => void;

  attachToContentDOM: (reactElement: HTMLDivElement) => void;

  constructor (props: PropsType) {
    super(props);
    this.update = () => {
      this.setState(() => ({
        nodeViewProps: props.nodeView.getNodeViewProps()
      }));
    };

    this.attachToContentDOM = (reactElement) => {
      if (!reactElement) {
        return;
      }
      const contentDOM = this.props.nodeView.contentDOM!;
      // Since we do not control how many times this callback is called
      // make sure it is not already mounted.
      if (!reactElement.contains(contentDOM)) {
        // If contentDOM happens to be mounted to someone else
        // remove it from there.
        if (contentDOM.parentNode) {
          contentDOM.parentNode.removeChild(contentDOM);
        }
        reactElement.appendChild(contentDOM);
      }
    };
    // So that we can directly update the nodeView without the mess
    // of prop forwarding.
    // What about updating the wrong nodeView ?
    // It is okay because a nodeView and this ReactComponent will always
    // have a 1:1 mapping. This is guaranteed because you use `nodeView` instance
    // to generate a react key. See the usage of this component in ./ReactEditor.js
    props.nodeViewUpdateStore.set(props.nodeView, this.update);
    this.state = { nodeViewProps: this.props.nodeView.getNodeViewProps() };
  }

  getChildren () {
    if (!this.props.nodeView.contentDOM) {
      return null;
    }

    // if (this.state.nodeViewProps.node.isInline) {
    //   return (
    //     // The bangle-nv-content is needed to keep the content take space
    //     // or else browsers will collapse it, making it hard to type
    //     <span
    //       className="bangle-nv-child-container"
    //       ref={this.attachToContentDOM}
    //     />
    //   );
    // }

    return (
      <div
        className='bangle-nv-child-container'
        ref={this.attachToContentDOM}
      />
    );
  }

  render () {
    const element = this.props.renderNodeViews({
      ...this.state.nodeViewProps,
      children: this.getChildren()
    });
    if (!element) {
      throw new Error(
        `renderNodeView must handle rendering for node of type "${this.state.nodeViewProps.node.type.name}"`
      );
    }
    return element;
  }
}
