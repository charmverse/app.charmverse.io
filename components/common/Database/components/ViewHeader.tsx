// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import { useRouter } from 'next/router';
import React, { useCallback, useState } from 'react';
import { FormattedMessage } from 'react-intl';
import { Board, IPropertyTemplate } from 'components/common/BoardEditor/focalboard/src/blocks/board';
import { BoardView } from 'components/common/BoardEditor/focalboard/src/blocks/boardView';
import { Card } from 'components/common/BoardEditor/focalboard/src/blocks/card';
import Button from 'components/common/BoardEditor/focalboard/src/widgets/buttons/button';
import ViewTabs from 'components/common/BoardEditor/focalboard/src/components/viewHeader/viewTabs';

import ModalWrapper from 'components/common/BoardEditor/focalboard/src/components/modalWrapper';

import FilterComponent from 'components/common/BoardEditor/focalboard/src/components/viewHeader/filterComponent';
import NewCardButton from 'components/common/BoardEditor/focalboard/src/components/viewHeader/newCardButton';
import ViewHeaderSortMenu from 'components/common/BoardEditor/focalboard/src/components/viewHeader/viewHeaderSortMenu';

type Props = {
    board: Board
    activeView: BoardView
    views: BoardView[]
    cards: Card[]
    addCard: () => void
    // addCardFromTemplate: (cardTemplateId: string) => void
    addCardTemplate: () => void
    editCardTemplate: (cardTemplateId: string) => void
    readonly: boolean
}

const ViewHeader = React.memo((props: Props) => {
  const router = useRouter();
  const [showFilter, setShowFilter] = useState(false);

  const { board, activeView, views, cards } = props;

  const withSortBy = activeView.fields.viewType !== 'calendar';

  const hasFilter = activeView.fields.filter && activeView.fields.filter.filters?.length > 0;

  const showView = useCallback((viewId) => {
    router.push({ pathname: router.pathname,
      query: {
        ...router.query,
        viewId: viewId || ''
      } }, undefined, { shallow: true });
  }, [router.query]);

  return (
    <div className='ViewHeader'>

      <ViewTabs
        views={views}
        readonly={props.readonly}
        showView={showView}
        board={board}
        activeView={activeView}
      />

      <div className='octo-spacer' />

      {/* Filter */}

      <ModalWrapper>
        <Button
          active={hasFilter}
          onClick={() => setShowFilter(true)}
        >
          <FormattedMessage
            id='ViewHeader.filter'
            defaultMessage='Filter'
          />
        </Button>
        {showFilter
                    && (
                    <FilterComponent
                      board={board}
                      activeView={activeView}
                      onClose={() => setShowFilter(false)}
                    />
                    )}
      </ModalWrapper>

      {/* Sort */}

      {withSortBy
                    && (
                    <ViewHeaderSortMenu
                      properties={board.fields.cardProperties}
                      activeView={activeView}
                      orderedCards={cards}
                    />
                    )}

      {/* Search */}

      {/* <ViewHeaderSearch /> */}

      {/* Options menu */}

      {!props.readonly && (
      <>
        {/* <ViewHeaderActionsMenu
              board={board}
              activeView={activeView}
              cards={cards}
            /> */}

        {/* New card button */}

        <NewCardButton
          addCard={props.addCard}
                  // addCardFromTemplate={props.addCardFromTemplate}
          addCardTemplate={props.addCardTemplate}
          editCardTemplate={props.editCardTemplate}
        />
      </>
      )}
    </div>
  );
});

export default ViewHeader;
