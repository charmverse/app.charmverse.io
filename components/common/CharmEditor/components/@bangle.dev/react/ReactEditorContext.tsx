import type { EditorView } from 'prosemirror-view';
import type { Dispatch, ReactNode, SetStateAction } from 'react';
import { createContext, useContext, useMemo, useState } from 'react';

// keep track of the focused page (may be different from what's in the URL or header)
type ICharmEditorView = {
  view: EditorView | null;
  setView: (view: EditorView) => void;
};

const CharmEditorView = createContext<Readonly<ICharmEditorView>>({
  view: null,
  setView: () => {}
});

export function CharmEditorViewProvider({ children }: { children: ReactNode }) {
  const [view, setView] = useState<EditorView | null>(null);

  const value: ICharmEditorView = useMemo(
    () => ({
      view,
      setView
    }),
    [view, setView]
  );

  return <CharmEditorView.Provider value={value}>{children}</CharmEditorView.Provider>;
}

export const useCharmEditorView = () => useContext(CharmEditorView);
