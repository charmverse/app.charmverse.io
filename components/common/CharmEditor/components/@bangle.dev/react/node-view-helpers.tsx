import { objectUid } from '@bangle.dev/utils';
import { log } from '@charmverse/core/log';
import { useEffect, useState } from 'react';
import type { RefObject } from 'react';
import { flushSync } from 'react-dom';

import type { NodeView, RenderHandlers } from '../core/node-view';
import { saveRenderHandlers } from '../core/node-view';

export const nodeViewUpdateStore = new WeakMap();

type NodeViewsUpdater = (nodeViewUpdateStore: NodeView[]) => NodeView[];
type UpdateNodeViewsFunction = (updater: NodeViewsUpdater) => void;

export const nodeViewRenderHandlers = (updateNodeViews: UpdateNodeViewsFunction): RenderHandlers => ({
  create: (nodeView, _nodeViewProps) => {
    log.debug('create node', objectUid.get(nodeView)); // , new Error().stack);
    updateNodeViews((nodeViews) => [...nodeViews, nodeView]);
  },
  update: (nodeView, _nodeViewProps) => {
    log.debug('update node', objectUid.get(nodeView));
    const updateCallback = nodeViewUpdateStore.get(nodeView);
    // If updateCallback is undefined (which can happen if react took long to mount),
    // we are still okay, as the latest nodeViewProps will be accessed whenever it mounts.
    if (updateCallback) {
      updateCallback();
    }
  },
  destroy: (nodeView) => {
    // log.debug('destroy', objectUid.get(nodeView));
    updateNodeViews((nodeViews) => nodeViews.filter((n) => n !== nodeView));
  }
});

export function useNodeViews(ref: RefObject<HTMLElement>) {
  const [nodeViews, setNodeViews] = useState<NodeView[]>([]);
  useEffect(() => {
    // save the renderHandlers in the dom to decouple nodeView instantiating code
    // from the editor. Since PM passing view when nodeView is created, the author
    // of the component can get the handler reference from `getRenderHandlers(view)`.
    // Note: this assumes that the pm's dom is the direct child of `editorRenderTarget`.
    let destroyed = false;
    saveRenderHandlers(
      ref.current!,
      nodeViewRenderHandlers((cb) => {
        if (!destroyed) {
          // make sure that updates run sequentially for prosemiror, or we get infinite recursion of node views being created and destroyed.
          // See also: https://github.com/ueberdosis/tiptap/pull/2985
          flushSync(() => {
            // use callback variant of setState to
            // always get freshest nodeViews.
            setNodeViews((_nodeViews) => cb(_nodeViews));
          });
        }
      })
    );
    return () => {
      destroyed = true;
    };
  }, [ref]);

  return nodeViews;
}
