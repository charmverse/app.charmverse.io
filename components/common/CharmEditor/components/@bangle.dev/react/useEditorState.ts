import { useState } from 'react';

import type { BangleEditorStateProps } from '../core/bangle-editor-state';
import { BangleEditorState } from '../core/bangle-editor-state';

export function useEditorState(props: BangleEditorStateProps) {
  if (props.plugins && typeof props.plugins !== 'function') {
    throw new Error('plugins error: plugins must be a function');
  }
  const [state] = useState(
    () =>
      // Instantiate the editorState once and keep using that instance
      // on subsequent renders.
      // Passing a callback in useState lazy calls the
      // functions on the first render and never again.
      new BangleEditorState(props)
  );
  return state;
}
