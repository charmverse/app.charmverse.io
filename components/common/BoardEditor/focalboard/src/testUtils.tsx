import type { PageMeta } from '@charmverse/core/pages';
import React from 'react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import { IntlProvider } from 'react-intl';
import type { Middleware } from 'redux';
import type { MockStoreEnhanced } from 'redux-mock-store';
import configureStore from 'redux-mock-store';
import { v4 } from 'uuid';

import { PagesContext } from 'hooks/usePages';
import type { Block } from 'lib/focalboard/block';
import { pageStubToCreate } from 'testing/generatePageStub';
import { AppThemeProvider } from 'theme/AppThemeProvider';

export const wrapIntl = (children?: React.ReactNode): JSX.Element => (
  <AppThemeProvider>
    <IntlProvider locale='en'>{children}</IntlProvider>
  </AppThemeProvider>
);
export const wrapDNDIntl = (children?: React.ReactNode): JSX.Element => {
  return <DndProvider backend={HTML5Backend}>{wrapIntl(children)}</DndProvider>;
};

export const wrapPagesProvider = (cardId: string | string[], children: React.ReactNode) => {
  const pages: Record<string, PageMeta> = {};

  const cardIds = Array.isArray(cardId) ? cardId : [cardId];

  cardIds.forEach((_cardId) => {
    pages[_cardId] = pageStubToCreate({ spaceId: v4(), createdBy: v4(), id: _cardId, path: _cardId }) as PageMeta;
  });

  return (
    <PagesContext.Provider
      value={
        {
          pages
        } as any
      }
    >
      {children}
    </PagesContext.Provider>
  );
};

export function mockDOM(): void {
  window.focus = jest.fn();
  document.createRange = () => {
    const range = new Range();
    range.getBoundingClientRect = jest.fn();
    range.getClientRects = () => {
      return {
        item: () => null,
        length: 0,
        [Symbol.iterator]: jest.fn()
      };
    };
    return range;
  };
}
export function mockMatchMedia(result: { matches: boolean }): void {
  // We check if system preference is dark or light theme.
  // This is required to provide it's definition since
  // window.matchMedia doesn't exist in Jest.
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(() => {
      return result;

      // return ({
      //     matches: true,
      // })
    })
  });
}

export function mockStateStore(middleware: Middleware[], state: unknown): MockStoreEnhanced<unknown, unknown> {
  const mockStore = configureStore(middleware);
  return mockStore(state);
}

export type BlocksById<BlockType> = { [key: string]: BlockType };

export function blocksById<BlockType extends Block>(blocks: BlockType[]): BlocksById<BlockType> {
  return blocks.reduce((res, block) => {
    res[block.id] = block;
    return res;
  }, {} as BlocksById<BlockType>);
}
