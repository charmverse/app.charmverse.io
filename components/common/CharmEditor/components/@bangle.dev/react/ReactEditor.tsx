import type {
  BangleEditorProps as CoreBangleEditorProps
} from '@bangle.dev/core';
import {
  BangleEditor as CoreBangleEditor
} from '@bangle.dev/core';
import type { Plugin, Transaction } from '@bangle.dev/pm';
import { EditorViewContext } from '@bangle.dev/react';
import { nodeViewUpdateStore, useNodeViews } from '@bangle.dev/react/node-view-helpers';
import { objectUid } from '@bangle.dev/utils';
import type { ReactNode, RefObject } from 'react';
import React, { useEffect, useImperativeHandle, useRef, useState } from 'react';
import reactDOM from 'react-dom';

import { useUser } from 'hooks/useUser';
import log from 'lib/log';
import { isTouchScreen } from 'lib/utilities/browser';

import { amendTransaction } from '../../suggestions/track/amendTransaction';

import { NodeViewWrapper } from './NodeViewWrapper';
import type { RenderNodeViewsFunction } from './NodeViewWrapper';

interface BangleEditorProps<PluginMetadata = any>
  extends CoreBangleEditorProps<PluginMetadata> {
  id?: string;
  children?: React.ReactNode;
  renderNodeViews?: RenderNodeViewsFunction;
  className?: string;
  style?: React.CSSProperties;
  onReady?: (editor: CoreBangleEditor<PluginMetadata>) => void;
  editorRef?: RefObject<HTMLDivElement>;
  // Components that should be placed underneath the editor
  placeholderComponent?: ReactNode;
  enableSuggestions?: boolean; // requires trackChanges to be true
  trackChanges?: boolean;
}

const updatePluginWatcher = (editor: CoreBangleEditor) => {
  return (watcher: Plugin, remove = false) => {
    if (editor.destroyed) {
      return;
    }

    let state = editor.view.state;

    const newPlugins = remove
      ? state.plugins.filter((p) => p !== watcher)
      : [...state.plugins, watcher];

    state = state.reconfigure({
      plugins: newPlugins
    });

    editor.view.updateState(state);
  };
};

export const BangleEditor = React.forwardRef<
  CoreBangleEditor | undefined,
  BangleEditorProps
>(
  (
    {
      id,
      state,
      children,
      focusOnInit = !isTouchScreen(),
      pmViewOpts,
      renderNodeViews,
      className,
      style,
      onReady = () => {},
      placeholderComponent,
      editorRef,
      enableSuggestions = false,
      trackChanges = false
    },
    ref
  ) => {
    const renderRef = useRef<HTMLDivElement>(null);
    const onReadyRef = useRef(onReady);
    const editorViewPayloadRef = useRef({
      state,
      focusOnInit,
      pmViewOpts,
      enableSuggestions
    });
    const [editor, setEditor] = useState<CoreBangleEditor>();
    const nodeViews = useNodeViews(renderRef);
    const { user } = useUser();

    if (enableSuggestions && !trackChanges) {
      log.error('CharmEditor: Suggestions require trackChanges to be enabled');
    }

    // set current
    editorViewPayloadRef.current.enableSuggestions = enableSuggestions;

    useImperativeHandle(
      ref,
      () => {
        return editor;
      },
      [editor]
    );

    useEffect(() => {
      const _editor = new CoreBangleEditor(
        renderRef.current!,
        editorViewPayloadRef.current
      );
      (_editor.view as any)._updatePluginWatcher = updatePluginWatcher(_editor);
      if (trackChanges) {
        (_editor.view as any)._props.dispatchTransaction = (transaction: Transaction) => {
          const view = _editor.view;
          const trackedUser = { id: user?.id ?? '', username: user?.username ?? '' };
          const trackedTr = amendTransaction(transaction, view.state, trackedUser, editorViewPayloadRef.current.enableSuggestions);
          const { state: newState } = view.state.applyTransaction(trackedTr);
          view.updateState(newState);
        };
      }
      onReadyRef.current(_editor);
      setEditor(_editor);
      return () => {
        _editor.destroy();
      };
    }, [ref]);

    if (nodeViews.length > 0 && renderNodeViews == null) {
      throw new Error(
        'When using nodeViews, you must provide renderNodeViews callback'
      );
    }

    return (
      <EditorViewContext.Provider value={editor?.view as any}>
        <div ref={editorRef} className='bangle-editor-core'>
          {editor ? children : null}
          <div ref={renderRef} id={id} className={className} style={style} />
          {editor ? placeholderComponent : null}
        </div>
        {nodeViews.map((nodeView) => {
          return reactDOM.createPortal(
            <NodeViewWrapper
              nodeViewUpdateStore={nodeViewUpdateStore}
              nodeView={nodeView}
              renderNodeViews={renderNodeViews!}
            />,
              nodeView.containerDOM!,
              objectUid.get(nodeView)
          );
        })}
      </EditorViewContext.Provider>
    );
  }
);
