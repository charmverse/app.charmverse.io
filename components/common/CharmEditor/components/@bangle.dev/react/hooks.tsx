import type { EditorView } from '@bangle.dev/pm';
import { Plugin, PluginKey } from '@bangle.dev/pm';
import { rafSchedule } from '@bangle.dev/utils';
import { useContext, useEffect, useState } from 'react';

import type { BangleEditorStateProps } from 'components/common/CharmEditor/components/@bangle.dev/core/bangle-editor-state';

import { BangleEditorState } from '../core/bangle-editor-state';

import { EditorViewContext } from './ReactEditor';

export function useEditorState(props: BangleEditorStateProps) {
  if (props.plugins && typeof props.plugins !== 'function') {
    throw new Error('plugins error: plugins must be a function');
  }

  const [state] = useState(
    () =>
      // Instantiate the editorState once and keep using that instance
      // on subsequent renders.
      // Passing a callback in useState lazy calls the
      // functions on the first render and never again.
      new BangleEditorState(props)
  );

  return state;
}

export function usePluginState(pluginKey: PluginKey, throttle = false) {
  const view = useEditorViewContext();
  const [state, setState] = useState(pluginKey.getState(view.state));

  useEffect(() => {
    let _setState = setState;
    if (throttle) {
      _setState = rafSchedule(setState);
    }
    const plugin = watcherPlugin(pluginKey, _setState);
    (view as any)._updatePluginWatcher(plugin);
    return () => {
      if (throttle) {
        (_setState as ReturnType<typeof rafSchedule>).cancel();
      }
      (view as any)._updatePluginWatcher(plugin, true);
    };
  }, [view, pluginKey, throttle]);

  return state;
}

export function useEditorViewContext(): EditorView {
  const context = useContext(EditorViewContext);
  if (!context) {
    throw new Error('useEditorViewContext must be used within a EditorViewContext');
  }
  return context;
}

function watcherPlugin(pluginKey: PluginKey, setState: (state: any) => void) {
  return new Plugin({
    key: new PluginKey(`withPluginState_${(pluginKey as any).key}`),
    view() {
      return {
        update(view, prevState) {
          const { state } = view;
          if (prevState === state) {
            return;
          }
          const newPluginState = pluginKey.getState(state);

          if (newPluginState !== pluginKey.getState(prevState)) {
            setState(newPluginState);
          }
        }
      };
    }
  });
}
