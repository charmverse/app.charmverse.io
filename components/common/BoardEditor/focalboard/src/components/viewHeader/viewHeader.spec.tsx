// @ts-nocheck

import { render } from '@testing-library/react';
import React from 'react';
import { Provider as ReduxProvider } from 'react-redux';

import '@testing-library/jest-dom';

import { TestBlockFactory } from '../../test/testBlockFactory';
import { mockStateStore, wrapIntl } from '../../testUtils';

import ViewHeader from './viewHeader';

const board = TestBlockFactory.createBoard();
const activeView = TestBlockFactory.createBoardView(board);
const card = TestBlockFactory.createCard(board);

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');

  return {
    ...originalModule,
    useRouteMatch: jest.fn(() => {
      return { url: '/board/view' };
    })
  };
});

describe('components/viewHeader/viewHeader', () => {
  const state = {
    users: {
      me: {
        id: 'user-id-1',
        username: 'username_1'
      }
    },
    searchText: {
    },
    boards: {
      current: board
    },
    cards: {
      templates: [card]
    },
    views: {
      views: {
        boardView: activeView
      },
      current: 'boardView'
    }
  };
  const store = mockStateStore([], state);
  test('return viewHeader', () => {
    const { container } = render(
      wrapIntl(
        <ReduxProvider store={store}>
          <ViewHeader
            board={board}
            activeView={activeView}
            views={[activeView]}
            cards={[card]}
            groupByProperty={board.fields.cardProperties[0]}
            addCard={jest.fn()}
            addCardFromTemplate={jest.fn()}
            addCardTemplate={jest.fn()}
            editCardTemplate={jest.fn()}
            readOnly={false}
          />
        </ReduxProvider>
      )
    );
    expect(container).toMatchSnapshot();
  });
  test('return viewHeader readonly', () => {
    const { container } = render(
      wrapIntl(
        <ReduxProvider store={store}>
          <ViewHeader
            board={board}
            activeView={activeView}
            views={[activeView]}
            cards={[card]}
            groupByProperty={board.fields.cardProperties[0]}
            addCard={jest.fn()}
            addCardFromTemplate={jest.fn()}
            addCardTemplate={jest.fn()}
            editCardTemplate={jest.fn()}
            readOnly={true}
          />
        </ReduxProvider>
      )
    );
    expect(container).toMatchSnapshot();
  });
});
