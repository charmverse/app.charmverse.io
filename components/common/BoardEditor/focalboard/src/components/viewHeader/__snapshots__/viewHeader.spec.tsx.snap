// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/viewHeader/viewHeader return viewHeader 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="ViewHeader "
    >
      <div
        class="MuiTabs-root css-5t1ahk-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-1dlgg0k-MuiButtonBase-root-MuiTab-root"
              data-view-id="view-id-1"
              draggable="true"
              role="tab"
              tabindex="-1"
              type="button"
              variant="text"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-4d711q-MuiTypography-root"
              >
                <span>
                  <svg
                    class="BoardIcon Icon"
                    fill="currentColor"
                    height="20"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      opacity="0.8"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M4 4H20V20H4V4ZM2 4C2 2.89543 2.89543 2 4 2H20C21.1046 2 22 2.89543 22 4V20C22 21.1046 21.1046 22 20 22H4C2.89543 22 2 21.1046 2 20V4ZM8 6H6V12H8V6ZM11 6H13V16H11V6ZM18 6H16V9H18V6Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </g>
                  </svg>
                  view title
                </span>
              </p>
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1br7sd5-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="view-actions MuiBox-root css-37urdo"
      >
        <button
          class="Button IconButton css-w119tu"
          style="width: 28px; height: 28px;"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeSmall css-fq32bd-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </button>
      </div>
      <div
        class="octo-spacer"
      />
      <div
        class="view-actions MuiBox-root css-0"
      >
        <button
          aria-haspopup="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation css-16efx8q-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Filter
        </button>
        <button
          aria-haspopup="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation css-16efx8q-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Sort
        </button>
        <div
          class="MuiBox-root css-jw1lpn"
        >
          <button
            class="Button IconButton css-w119tu"
            style="width: 32px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-qolhez-MuiSvgIcon-root"
              data-testid="MoreHorizOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </button>
        </div>
        <div
          class="MuiButtonGroup-root MuiButtonGroup-contained MuiButtonGroup-disableElevation css-76ejyv-MuiButtonGroup-root"
          role="group"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation css-1pjzawa-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiTypography-root MuiTypography-subtitle1 css-15znijn-MuiTypography-root"
            >
              New
            </div>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation css-1lvcm06-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-qolhez-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </span>
</div>
`;

exports[`components/viewHeader/viewHeader return viewHeader readonly 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="ViewHeader "
    >
      <div
        class="MuiTabs-root css-5t1ahk-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-1dlgg0k-MuiButtonBase-root-MuiTab-root"
              data-view-id="view-id-1"
              draggable="true"
              role="tab"
              tabindex="-1"
              type="button"
              variant="text"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-4d711q-MuiTypography-root"
              >
                <span>
                  <svg
                    class="BoardIcon Icon"
                    fill="currentColor"
                    height="20"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      opacity="0.8"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M4 4H20V20H4V4ZM2 4C2 2.89543 2.89543 2 4 2H20C21.1046 2 22 2.89543 22 4V20C22 21.1046 21.1046 22 20 22H4C2.89543 22 2 21.1046 2 20V4ZM8 6H6V12H8V6ZM11 6H13V16H11V6ZM18 6H16V9H18V6Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </g>
                  </svg>
                  view title
                </span>
              </p>
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1br7sd5-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="octo-spacer"
      />
      <div
        class="view-actions MuiBox-root css-0"
      />
    </div>
  </span>
</div>
`;
